infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/project/OpenShiftProjectFactory.java

```diff
@@ -116,6 +117,19 @@ public OpenShiftProject create(String workspaceId) throws InfrastructureExceptio
     return osProject;
   }
 
+  @VisibleForTesting
+  @Override
+  protected String evalNamespaceName(String workspaceId, Subject currentUser)
+      throws NotFoundException, ServerException, InfrastructureException, ConflictException,
+          ValidationException {
+    return super.evalNamespaceName(workspaceId, currentUser);
+  }
+
+  @Override
+  protected boolean checkNamespaceExists(String namespaceName) throws InfrastructureException {
+    return clientFactory.createOC().projects().withName(namespaceName).get() != null;
```

Comment:
It never is null. OpenShift returns forbidden if project does not exist.
![Screenshot_20191010_115105](https://user-images.githubusercontent.com/5887312/66553694-4346bb80-eb54-11e9-9b33-daa7d73cf629.png)

Reply:
See https://github.com/eclipse/che/pull/14541/files#diff-fd303707b626771b58ee33d25ddc3afaR136

Reply:
In your case checking 403 - is enough, since if the project exists but not available for user - it means that he is not able to run workspace there.
In my case it would be better to investigate how OpenShift console handles it, if I'm not mistaken it distinct (during new project creation) when project does not exist and not available.

Reply:
Thanks! Fixed.